src/fdc3/dotnet/AppDirectory/src/MorganStanley.ComposeUI.AppDirectory/DependencyInjection/ServiceCollectionAppDirectoryExtensions.cs

```diff
@@ -25,7 +25,8 @@ public static class ServiceCollectionAppDirectoryExtensions
     public static IServiceCollection AddFdc3AppDirectory(this IServiceCollection serviceCollection)
     {
         serviceCollection.AddSingleton<IAppDirectory, AppDirectory>();
-        serviceCollection.AddSingleton<IModuleCatalog, Fdc3ModuleCatalog>();
+        serviceCollection.AddSingleton<Fdc3ModuleCatalog>();
```

Comment:
We can't reach the Fdc3ModuleCatalog when using the older stament.
You can see here as well:
https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/App.xaml.cs#L229-L240

We cant reach this kind of `ModuleCatalog` if we want to. This was when I did some experimentation on the examples, cause right now the FDC3 examples(grid and chart) does not work properly through passing them as cmdline arguments - the scripts are not injected which is understandable.
but basically what happens when we open the url:https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/App.xaml.cs#L227-L240 and https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/Properties/launchSettings.json#L5 then we do not find the given generated name in the appdir.(https://github.com/morganstanley/ComposeUI/blob/main/src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/Fdc3StartupAction.cs#L42); 

I am not sure if this is the intended behavior in case of a FDC3 enabled app (enabled through cmdline property but the script is not injected). I did not fix this in this PR as it needs more discussion. But I kept this change to keep it consistent with the other registration that you can find here: 
https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/App.xaml.cs#L182-L183

Reply:
We can't reach the Fdc3ModuleCatalog when using the older stament.
You can see here as well:
https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/App.xaml.cs#L229-L240

We cant reach this kind of `ModuleCatalog` if we want to. This was when I did some experimentation on the examples, cause right now the FDC3 examples(grid and chart) does not work properly through passing them as cmdline arguments - the scripts are not injected which is understandable.
but basically what happens when we open the url:https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/App.xaml.cs#L227-L240 and https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/Properties/launchSettings.json#L5 then we do not find the given generated name in the appdir.(https://github.com/morganstanley/ComposeUI/blob/main/src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/Fdc3StartupAction.cs#L42); 

I am not sure if this is the intended behavior in case of a FDC3 enabled app (enabled through cmdline property but the script is not injected). I did not fix this in this PR as it needs more discussion. But I kept this change to keep it consistent with the other registration that you can find here: 
https://github.com/morganstanley/ComposeUI/blob/main/src/shell/dotnet/Shell/App.xaml.cs#L182-L183

Reply:
I understand, thanks :)
We should take a look at this in the future, but I agree this is not the PR for that.